app/objects/c_ability.py

```diff
@@ -118,16 +123,28 @@ def find_executors(self, platform, names):
                 continue
             seen_names.add(name)
 
-            matched_executor = self.find_executor(platform, name)
-            if matched_executor:
-                executors.append(matched_executor)
+            if f'{platform}-{name}' in self._executor_map:
+                executors.append(self.find_executor(platform, name))
         return executors
 
+    def add_executors(self, executors):
+        """Create executor map from list of executor objects"""
+        for executor in executors:
+            self.add_executor(executor)
+
     def add_executor(self, executor):
-        existing_executor = self.find_executor(executor.platform, executor.name)
-        if existing_executor:
-            self.executors.remove(existing_executor)
-        self.executors.append(executor)
+        """Add executor to map
+
+        If the executor exists, delete the current entry and add the
+            new executor to the bottom for FIFO
+        """
+        unique_name = f'{executor.platform}-{executor.name}'
```

Comment:
What do you think about using a tuple as a key rather than the formatted string? That is...

```
self._executor_map[e.platform, e.name] 
```

IMO it cleans up the code a bit. Either way, it might be nice to consolidate the key generation code into a helper method so you don't need to copy around the logic:

```python
def _make_executor_map_key(self, executor):
    return (executor.platform, executor.name)  # or the f-string
```